libraries/AP_HAL_Linux/Util_RPI.cpp

```diff
@@ -27,47 +27,67 @@ using namespace Linux;
 
 UtilRPI::UtilRPI()
 {
-    _check_rpi_version_by_rev();
+    _get_peripheral_base ();
 }
 
-int UtilRPI::_check_rpi_version_by_rev()
-{
-    // assume 2 if unknown
-    _rpi_version = 2;
-    const char *SOC[]= {"Broadcom BCM2835","Broadcom BCM2836","Broadcom BCM2837","Broadcom BCM2711"};
-    const char *revision_file_ = "/proc/device-tree/system/linux,revision";
-    uint8_t revision[4] = { 0 };
-    uint32_t cpu = 0;
-    FILE *fd;
 
-    if ((fd = fopen(revision_file_, "rb")) == NULL) {
-        printf("Can't open '%s'\n", revision_file_);
-    }
-    else {
-        if (fread(revision, 1, sizeof(revision), fd) == 4) {
-            cpu = (revision[2] >> 4) & 0xf;
-            
-            _rpi_version = cpu;
+// 
+// previous appraoch was using /proc/device-tree/system/linux,revision
+// now we use /proc/device-tree/soc/ranges see: https://forums.raspberrypi.com//viewtopic.php?t=244031
+uint32_t UtilRPI::_get_peripheral_base() 
```

Comment:
This is kind of missing the point of the method now!

```suggestion
void UtilRPI::detect_board_type() 
```

Reply:
then I suggest to rename it to **_get_board_type_using_peripheral_base** because this is what it does .
there is a function called **detect_board_type** that returns LINUX_BOARD_TYPE